File: LogGenerator/src/main/java/com/loggenerator/LogGenerator.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */
package com.loggenerator;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.List;
import java.util.Random;
import java.util.concurrent.TimeUnit;

public class LogGenerator {

    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
    
    
    private static String getRANDOMLines(Random random, String file) throws IOException{
        List<String> lines = Files.readAllLines(Paths.get(file));
        return lines.get(random.nextInt(lines.size()));
    }
    
    
    private static String generateDate(Random random) {
        Calendar calendar = Calendar.getInstance();

        int year = random.nextInt(2) + 2023;
        int month = random.nextInt(12);
        int day = random.nextInt(calendar.getActualMaximum(Calendar.DAY_OF_MONTH)) + 1;

        calendar.set(year, month, day);

        SimpleDateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");
        String data = dateFormat.format(calendar.getTime());
        return data;
    }

    public static String generateHours(Random random) {
        int hora = random.nextInt(24);
        int minuto = random.nextInt(60);

        String horaFormatada = String.format("%02d:%02d", hora, minuto);
        return horaFormatada;
    }

    private static String generateCPF(Random random) {
        String cpf = "";        
        for (int i = 0; i < 3; i++) {
            for (int j = 0; i < 3; i++) {
                cpf += String.format("%d", random.nextInt(10));
            }  
            if(i != 2)
                cpf += String.format(".");
        }
        cpf += String.format("-%d%d", random.nextInt(10), random.nextInt(10));
        return cpf;
    }

    public static String generateRandomId() {
        String validChars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
        StringBuilder sb = new StringBuilder();
        Random random = new Random();
        for (int i = 0; i < 7; i++) {
            sb.append(validChars.charAt(random.nextInt(validChars.length())));
        }
        return sb.toString();
    }

    private static void grammarPrepare(Random random) throws InterruptedException, IOException {
        for (int i = 0; i < 100; i++) {
            String cpf = generateCPF(random);
            String action = getRANDOMLines(random, "LogGenerator/text_files/action.txt");
            String message = getRANDOMLines(random, "LogGenerator/text_files/messages.txt");
            String to_user = generateCPF(random);
            String id = generateRandomId();
            String fileName = getRANDOMLines(random, "LogGenerator/text_files/files.txt");
            String content = getRANDOMLines(random, "LogGenerator/text_files/contents.txt");
            String name = getRANDOMLines(random, "LogGenerator/text_files/names.txt");
            String amount = "$" + String.valueOf((random.nextInt(99901) + 10) * 100) + ",00";
            String date = generateDate(random);
            String hours = generateHours(random);

            message = message
                .replace("{CUSTOMER_ID}", id)
                .replace("{TIME}", hours)
                .replace("{DATE}", date)
                .replace("{AMOUNT}", amount)
                .replace("{USER_NAME}", name)
                .replace("{SENDER_NAME}", name)
                .replace("{CUSTOMER_NAME}", name)
                .replace("{CPF}", "<s>" + cpf + "</s>")
                .replace("{CONTENT}", content)
                .replace("{FILE_NAME}", fileName)
                .replace("{ID}", id)
                .replace("{TO_USER}", "<s>" + to_user + "</s>")
                .replace("{ACTION}", action)
                .replace("{DATETIME}", DATE_FORMAT.format(new Date()));

            System.out.println(message);
        }
    }
    
    private static void printMessages(Random random, int n) throws IOException, InterruptedException{
        for (int i = 0; i < n; i++) {
            String cpf = generateCPF(random);
            String action = getRANDOMLines(random, "LogGenerator/text_files/action.txt");
            String message = getRANDOMLines(random, "LogGenerator/text_files/messages.txt");
            String to_user = generateCPF(random);
            String id = generateRandomId();
            String fileName = getRANDOMLines(random, "LogGenerator/text_files/files.txt");
            String content = getRANDOMLines(random, "LogGenerator/text_files/contents.txt");
            String name = getRANDOMLines(random, "LogGenerator/text_files/names.txt");
            String amount = "$" + String.valueOf((random.nextInt(99901) + 10) * 100) + ",00";
            String date = generateDate(random);
            String hours = generateHours(random);

            message = message
                .replace("{CUSTOMER_ID}", id)
                .replace("{TIME}", hours)
                .replace("{DATE}", date)
                .replace("{AMOUNT}", amount)
                .replace("{USER_NAME}", name)
                .replace("{SENDER_NAME}", name)
                .replace("{CUSTOMER_NAME}", name)
                .replace("{CPF}",  cpf )
                .replace("{CONTENT}", content)
                .replace("{FILE_NAME}", fileName)
                .replace("{ID}", id)
                .replace("{TO_USER}", to_user )
                .replace("{ACTION}", action)
                .replace("{DATETIME}", DATE_FORMAT.format(new Date()));

            System.out.println(message);
            TimeUnit.SECONDS.sleep(1);
        }
    }
    
    public static void main(String[] args) throws InterruptedException, IOException {
        System.setOut(new CustomPrintStream(System.out));

        Random random = new Random();
        grammarPrepare(random);
        int numMessages = 500; 
        printMessages(random, numMessages);
    }
}

File: Lushu/src/main/kotlin/lushu/LogGenerator/LogGenerator.kt
package lushu.LogGenerator

import java.io.File
import java.nio.file.Paths
import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Date
import java.util.Random
import java.util.concurrent.TimeUnit

class LogGenerator(
    private val logExamplesDir: String
) {
    private val random = Random()
    private val datefmt = SimpleDateFormat("yyyy-MM-dd HH:mm:ss")

    private val files = File(logExamplesFpath("files.txt")).readLines()
    private val contents = File(logExamplesFpath("contents.txt")).readLines()
    private val names = File(logExamplesFpath("names.txt")).readLines()
    private val actions = File(logExamplesFpath("actions.txt")).readLines()
    private val messages = File(logExamplesFpath("messages.txt")).readLines()

    fun run(numLogs: Int, sleepTime: Long = 0.toLong()) {
        for (i in 0 until numLogs) {
            dispatchLog()
            if (sleepTime > 0) {
                TimeUnit.MILLISECONDS.sleep(sleepTime)
            }
        }
    }

    private fun generateDate(): String {
        val calendar: Calendar = Calendar.getInstance()
        val year: Int = random.nextInt(2) + 2023
        val month: Int = random.nextInt(12)
        val day: Int = random.nextInt(calendar.getActualMaximum(Calendar.DAY_OF_MONTH)) + 1
        calendar.set(year, month, day)
        return SimpleDateFormat("dd/MM/yyyy").format(calendar.getTime())
    }

    private fun generateHours(): String {
        return String.format("%02d:%02d", random.nextInt(24), random.nextInt(60))
    }

    private fun generateCPF(): String {
        var cpf: String = ""
        for (i in 0..2) {
            for (j in 0..2) {
                cpf += random.nextInt(10).toString()
            }
            if (i != 2) {
                cpf += "."
            }
        }
        cpf += String.format("-%d%d", random.nextInt(10), random.nextInt(10))
        return cpf
    }

    private fun generateRandomId(): String {
        val validChars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
        var s = ""
        for (i in 0..7) {
            s += validChars[random.nextInt(validChars.length)]
        }
        return s
    }

    private fun generateRandomAmount(): String {
        return "\$${(random.nextInt(99901) + 10) * 100},00"
    }

    private fun logExamplesFpath(relativePath: String): String {
        return Paths.get(logExamplesDir, relativePath).toAbsolutePath().toString()
    }

    private fun dispatchLog() {
        val fileName = files.random()
        val content = contents.random()
        val name = names.random()
        val action = actions.random()

        val cpf = generateCPF()
        val to_user = generateCPF()
        val id = generateRandomId()
        val amount = generateRandomAmount()
        val date = generateDate()
        val hours = generateHours()

        val message = messages.random()
            .replace("{CUSTOMER_ID}", id)
            .replace("{TIME}", hours)
            .replace("{DATE}", date)
            .replace("{AMOUNT}", amount)
            .replace("{USER_NAME}", name)
            .replace("{SENDER_NAME}", name)
            .replace("{CUSTOMER_NAME}", name)
            .replace("{CPF}", cpf)
            .replace("{CONTENT}", content)
            .replace("{FILE_NAME}", fileName)
            .replace("{ID}", id)
            .replace("{TO_USER}", to_user)
            .replace("{ACTION}", action)
            .replace("{DATETIME}", datefmt.format(Date()))

        println(message)
    }
}
